### ISSUE TITLE
`smesh` will fail the build with gcc-11: error: ISO C++17 does not allow dynamic exception specifications

### ISSUE THREAD

**Issue by trofi** [NixOS/nixpkgs#143973]
`gcc-11` switched to `gnu++17 ` standard by default:

```
$ nix-build -E 'with import <nixpkgs> {}; smesh.override { stdenv = gcc11Stdenv; }'
...
In file included from /build/source/src/SMDS/SMDS_Mesh.cpp:29:
/build/source/inc/SMDS_Mesh.hxx:501:55: error: ISO C++17 does not allow dynamic exception specifications
  501 |   static int CheckMemory(const bool doNotRaise=false) throw (std::bad_alloc);
      |                                                       ^~~~~
/build/source/src/SMDS/SMDS_Mesh.cpp:62:51: error: ISO C++17 does not allow dynamic exception specifications
   62 | int SMDS_Mesh::CheckMemory(const bool doNotRaise) throw (std::bad_alloc)
      |                                                   ^~~~~
```

### Notify maintainers

@gebner

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: smesh
```

**Comment by trofi** [comment]
Closing as INVALID.

**Comment by vcunat** [comment]
Switching to gcc 11 again.  Fixed in 1ca464f98f.
